Cleanups/type hints + documentation

Quix0r opened the pull request:
Changes:
- added missing type-hints
- added documentation
- removed parameter $baseURL while it can be locally get the same way but lesser parameter
- used `ActivityNamespace::SALMON_ME` instead of direct string
- changed double-quotes to single

Quix0r committed:
Changes:
- added type-hint for $type parameter
- documented method
- unsupported verbs logged

Quix0r committed:
Changes:
- methods should ALWAYS start lower-case, not upper
- documented a bit more

Quix0r committed:
Changes:
- added some missing type-hints
- added some documentation
- added some empty lines to `@return` statements (code-style convention)

Quix0r committed:
Changes:
- added missing type-hints
- added documentation
- removed parameter $baseURL while it can be locally get the same way but
  lesser parameter
- used ActivityNamespace::SALMON_ME
- changed double-quotes to single

MrPetovan commented on src/Module/Xrd.php:
```diff
@@ -153,8 +153,9 @@ private static function printSystemJSON(array $owner)
-	private static function printJSON($alias, $baseURL, $owner, $avatar)
+	private static function printJSON(string $alias, array $owner, array $avatar)
 	{
+		$baseURL = DI::baseUrl()->get();
```
Since we're depending on a `BaseModule`, this can be simplified further:
```suggestion
		$baseURL = $this->baseUrl->get();
```

MrPetovan commented on src/Module/Xrd.php:
```diff
@@ -234,8 +235,9 @@ private static function printJSON($alias, $baseURL, $owner, $avatar)
-	private static function printXML($alias, $baseURL, $user, $owner, $avatar)
+	private static function printXML(string $alias, array $user, array $owner, array $avatar)
 	{
+		$baseURL = DI::baseUrl()->get();
```
```suggestion
		$baseURL = $this->baseUrl->get();
```

MrPetovan reviewed (changes requested):
Did you run this code on your node? I'm weary the addition of type-hints to `Notify->storeAndSend()` might lead to fatal errors with null values.

Quix0r committed:
Changed:
- used $this->baseurl instead of DI::baseurl() (@MrPetovan)

Quix0r commented:
Yes, I run this code on my instance.

annando commented on src/Navigation/Notifications/Repository/Notify.php:
```diff
@@ -479,7 +482,8 @@ function createFromArray($params)
 						break;
-					case "SYSTEM_DB_UPDATE_FAIL":
+
+					case 'SYSTEM_DB_UPDATE_FAIL': // @TODO Unused (only here)
```
`@TODO` should be used to indicate code parts that need rework or could be enhanced in the future. I don't a reason for a `@TODO` here. Some `// currently unused` would be more appropriate, I think.

Quix0r replied:
The disadvantage is that it (maybe this is a weak example) gets lost and ignored over time.

annando replied:
Because of the over usage of `@TODO` I now tend to simply ignore them absolutely.

Quix0r replied:
That' not my problem, or is it?

MrPetovan replied:
Unfortunately, it is your problem. You're writing code not only for the machines that will execute it, but also for the humans working on it alongside you.

Quix0r committed:
Changed:
- double-quotes to single
- more space around concated strings
- removed TODO as there is sadly no IOException in PHP
